test/integration_test/applicationbackup_test.go

```diff
@@ -0,0 +1,472 @@
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testKey = "mysql-1-pvc"
+	appKey  = "mysql"
+
+	applicationBackupScheduleRetryInterval = 10 * time.Second
+	applicationBackupScheduleRetryTimeout  = 3 * time.Minute
+)
+
+func testApplicationBackup(t *testing.T) {
+	t.Run("applicationBackupRestoreTest", applicationBackupRestoreTest)
+	t.Run("scheduleTests", applicationBackupScheduleTests)
+}
+
+func triggerBackupRestoreTest(
+	t *testing.T,
+	appBackupKey []string,
+	appRestoreKey []string,
+) {
+	var err error
+	var ctxs []*scheduler.Context
+	ctx := createApp(t, appKey)
+	ctxs = append(ctxs, ctx)
+	logrus.Infof("App created %v. Starting backup.", ctx.GetID())
+
+	// Create backuplocation here programatically using config-map that contains name of secrets to be used, passed from the CLI
+	err = createBackupLocation(t, appKey+"-backup-location", ctx.GetID(), storkv1.BackupLocationS3, "secret-config")
+	require.NoError(t, err, "Error creating backuplocation")
+
+	// Backup application
+	err = schedulerDriver.AddTasks(ctxs[0],
+		scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	require.NoError(t, err, "Error backing-up apps")
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for back-up to complete.")
+
+	logrus.Infof("Backup completed. Starting Restore.")
+
+	// Restore application
+	err = schedulerDriver.AddTasks(ctxs[0],
+		scheduler.ScheduleOptions{AppKeys: appRestoreKey})
+	require.NoError(t, err, "Error restoring apps")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for restore to complete.")
+
+	logrus.Infof("Restore completed.")
+
+	// Validate applications after restore
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for restore to complete.")
```

Comment:
Do we also need to verify whether volume driver has restored objects and all app elements are created

Reply:
`WaitForRunning` makes sure the app is up and running which includes all the specs and volumes.
Also, this is showing up in the diff because I renamed the file. This test was already present.